sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/CloudEvent.java

```diff
@@ -40,22 +40,35 @@
 
     private static final ClientLogger logger = new ClientLogger(CloudEvent.class);
 
-    private boolean parsed = false;
-
-    private static final JsonSerializer deserializer = new JacksonJsonSerializerBuilder()
-        .serializer(new JacksonAdapter().serializer() // this is a workaround to get the FlatteningDeserializer
-            .registerModule(new JavaTimeModule())) // probably also change this to DateTimeDeserializer when/if it
-        .build();                                  // becomes public in core
+    /**
+     * Create an instance of CloudEvent. The source and type are required fields to publish.
+     * @param source a URI identifying the origin of the event. It can't be null or empty.
+     * @param type   the type of event, e.g. "Contoso.Items.ItemReceived". It can't be null or empty.
+     * @param data the payload of this event. Set to null if your event doesn't have the data payload.
+     *             It will be serialized as a String if it's a String, or application/json if it's not a String.
+     */
+    public CloudEvent(String source, String type, Object data) {
+        this(source, type);
+        this.setData(data);
+    }
 
     /**
-     * Create an instance of a CloudEvent. The source and type are required fields to publish.
+     * Create an instance of CloudEvent. The source and type are required fields to publish.
      * @param source a URI identifying the origin of the event.
      * @param type   the type of event, e.g. "Contoso.Items.ItemReceived".
+     * @param data the payload in bytes of this event. It will be serialized to Base64 format.
+     * @param dataContentType the type of the data.
      */
-    public CloudEvent(String source, String type) {
+    public CloudEvent(String source, String type, byte[] data, String dataContentType) {
+        this(source, type);
+        this.setDataBase64(data, dataContentType);
+    }
+
+    private CloudEvent(String source, String type) {
         if (CoreUtils.isNullOrEmpty(source)) {
             throw logger.logExceptionAsError(new IllegalArgumentException("Source cannot be null or empty"));
-        } else if (CoreUtils.isNullOrEmpty(type)) {
+        }
+        if (CoreUtils.isNullOrEmpty(type)) {
             throw logger.logExceptionAsError(new IllegalArgumentException("type cannot be null or empty"));
```

Comment:
nit: casing,  `t` in `type` should be capitol `Type can not be null or empty.`

Reply:
I changed to 'type' because the param name is type.